src/controllers/recipesController.js

```diff
@@ -0,0 +1,84 @@
+const BaseController = require('./baseController')
+
+class RecipesController extends BaseController {
+  constructor(a) {
+    super(a)
+  }
+
+  async index(req, res) {
+    const recipes = await this.models.recipes.findAll({
+      include: this.models.users,
+    })
+
+    var output = new Set()
+    recipes.forEach((item) => {
+      output.add(item.type)
+    })
+
+    const types = output
+
+    res.render('pages/recipes/index.html.ejs', {
+      recipes,
+      types,
+      context: await this.viewContext(req),
+    })
+  }
+
+  async show(req, res) {
+    const recipe = await this.models.recipes.findOne({
+      where: {
+        id: req.params.id,
+      },
+
+      include: [
+        this.models.users,
+        this.models.recipe_ratings,
+        {
+          model: this.models.comments,
+          include: this.models.users,
+        },
+      ],
+      order: [[this.models.comments, 'createdA', 'DESC']],
+    })
+
+    if (recipe) {
+      res.render('pages/recipes/show.html.ejs', {
+        recipe,
+        totalRating: await recipe.getTotalRating(),
+        myRating: req.user && recipe.recipe_ratings.find((recipeRating) => recipeRating.userId === req.user.id)?.rating,
+        recipesUrl: `${this.utils.fullBaseUrl}/recipes`,
+        context: await this.viewContext(req),
+      })
+    } else {
+      res.sendStatus(404)
+    }
+  }
+
+  async rate(req, res) {
+    const recipe = await this.models.recipes.findOne({ where: { id: req.params.id } })
+    const recipeRating = await this.models.recipe_ratings.findOne({
+      where: {
+        userId: req.user.id,
+        recipeId: req.params.id,
+      },
+    })
+
+    if (!recipe) {
+      res.sendStatus(404)
+    } else {
```

Comment:
Geriau naudoti `return`:
```suggestion
    if (!recipe) {
      res.sendStatus(404)
      return
    }
```